### PR TITLE
Assembly update

### LINKED ISSUE #11406: Emptying a bag onto the floor via drag doesn't trigger mousetraps
This basically allows you to bypass moustrap bombs pretty easily.

### LINKED ISSUE #11521: Placing an APC will have it pointing the opposite direction.
What it says on the tin. You end up with a floating APC over the floor when you try to place it on a wall.

### PR DESCRIPTION
**Pull request by Core0verload**
Attention! This PR contains **PURE EVIL**.
- Ported health sensor assembly from NTstation. Suicide bombing is fun!
- Health sensors can be attached to wires. 
- Flash is now an assembly. Flashes activated by assemblies are not very powerful, they do a weak radius flash, like flash's attack_self.
- You can attach mousetraps to door wires. Mousetrap will trigger when someone will try to open door's wires window.
- Emptying storages will trigger mousetraps in it. Fixes #11406.
- Changed the way assemblyholder icon works. Instead of using left and right icon for every assembly and every overlay, it uses only left icons and mirror transform matrix.

Fixes #11521. It's unrelated to assemblies, but an extra fix wouldn't hurt, right?

### CODE REVIEW

**Comment by ike709** [comment]
:+1: x :100: 

Health sensors were a great NT feature that I'm surprised we haven't ported yet

File: code/modules/assembly/health.dm [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+/obj/item/device/assembly/health/process()
+	if(!scanning || !secured)
+		return
+
+	var/atom/A = src
+	if(connected && connected.holder)
+		A = connected.holder
+
+	for(A, A && !ismob(A), A=A.loc);
```
Comment (RemieRichards) [inline code/modules/assembly/health.dm]: Won't this mean if a mob with a health assembly is inside another mob, it'll be checking the mob that ate them's health?

**Comment by Fox-McCloud** [comment]
Suicide vests when?

**Comment by Lumipharon** [comment]
>porting health assemblies

My rock is cock hard.

**Comment by optimumtact** [comment]
>You can attach mousetraps to door wires. Mousetrap will trigger when someone will try to open door's wires window.

hehehe

**Comment by PKPenguin321** [comment]
:+1: :+1: :+1: good stuff

**Comment by WJohn** [comment]
Still porting things from NTstation, nearly a year later.

File: code/modules/assembly/health.dm (hunk 1 above)
Comment (Core0verload) [inline code/modules/assembly/health.dm]: No. It works like get_turf - loops in .loc vars until finds any mob.

Comment (RemieRichards) [inline code/modules/assembly/health.dm]: Ah yes, true enough, I wasn't reading, 12 hours ago would put it at the exact hour I woke up, probably within 10 minutes haha.

**Comment by Supermichael777** [comment]
>mouse trap break in shock trap

I love it

**Comment by Core0verload** [comment]
Conflicts are resolved, commits are squished, this pull is ready to merge!

**Pull request merged**